app/models/watching_transaction.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require 'ethereum_api'
+
+class WatchingTransaction < ApplicationRecord
+  before_create :set_uuid
+
+  belongs_to :user
+  has_many :transactions, class_name: 'WatchingTransaction', foreign_key: 'group_id'
+  belongs_to :original, class_name: 'WatchingTransaction', foreign_key: 'group_id', optional: true
+
+  validates :transaction_object,
+            presence: true
+  validates :signed_transaction,
+            presence: true
+  validates :txhash,
+            presence: true,
+            uniqueness: true
+
+  def set_uuid
+    self.id = SecureRandom.uuid
+    self.group_id = id unless group_id
+  end
+
+  def transaction_object
+    JSON.parse(super)
+  end
+
+  def transaction_object=(value)
+    super(JSON.generate(value))
+  end
+
+  def txhash=(value)
+    super(value&.downcase)
+  end
+
+  class << self
+    def watch(user, transaction_object, signed_transaction, group_id)
+      tx = WatchingTransaction.new(
+        user: user,
+        transaction_object: transaction_object,
+        signed_transaction: signed_transaction,
+        txhash: transaction_object['transactionHash'],
+        group_id: group_id
+      )
+
+      return [:invalid_data, tx.errors] unless tx.valid?
+      return [:database_error, tx.errors] unless tx.save
+
+      [:ok, tx]
+    end
+
+    def resend
```

Comment:
Make sure this is also tested

Reply:
The algorithm should be something like so:

```
For every transaction group
  If any transaction in the group is missing
    If the latest in the group is dropped, 
      Resend that one
  If any transaction in the group is mined
    Delete the whole group
```

Reply:
Right now it tries to get tx (every tx in the table) by hash and if it cannot be fetched it deletes tx if it's not the latest in the group or resends tx if it's latest one. If tx is fetched and mined, then it deletes all txs from that group.

Scenarios to test:
1. Resend latest tx in the group (rest txs should be destroyed)
2. Destroy group if at least one tx from it is mined

Local ganache network is required here or just mocking of the EthereumApi. What do you think?

Reply:
Preferably `WebMock` but local development testing you should use a local ganache network to capture possible responses for WebMock.